### PR TITLE
ensure MCBP session closure when topology changes

### PR DESCRIPTION
**Pull request by avsej**

### CODE REVIEW

**Review by davidkelly — APPROVED** [review]
LGTM

**Comment by brett19** [comment]
I wonder if we should modify certain things like mcbp_session to be non-copyable?

File: couchbase/bucket.hxx [hunk 1]
```diff
@@ -213,7 +214,7 @@ class bucket : public std::enable_shared_from_this<bucket>
         } else {
             new_session = std::make_shared<io::mcbp_session>(client_id_, ctx_, origin_, name_, known_features_);
         }
-        new_session->bootstrap([self = shared_from_this(), new_session, h = std::forward<Handler>(handler)](
+        new_session->bootstrap([self = shared_from_this(), new_session = std::move(new_session), h = std::forward<Handler>(handler)](
```
Comment (avsej) [inline couchbase/bucket.hxx:217]: yes, it is not necessary here, but above it is the last usage of the pointer, so we can safely move it into the closure

**Comment by avsej** [comment]
@brett19 I've updated the patch, and removed questionable `std::move`. Now the only move left in the place, where we know for sure the pointer is not used anymore

**Review by brett19 — APPROVED** [review]

**Pull request merged**